### PR TITLE
Add profiling to Python2 code. 

### PR DESCRIPTION
**Pull request by snim2**
Adds a menu item for the profiler. Callls a profiler from the Python2 stdlib and adds anotations to each function definition. Annotations are bold until the code is 'stale' -- i.e. it has been modified.

Things that need to be done before this PR can be merged:
- ~~Disable all profiling / tooling actions when the language of the current language does not support profiling~~
- ~~Run profilers and other tools in a worker thread (helpful when editing long-running programs)~~ (worker thread in `eco.py`)
- ~~Display a throbber while collecting profiling information (helpful when editing long-running programs)~~
- ~~When profiling, any output to STDOUT should appear in the Eco console~~

### CODE REVIEW

**Comment by snim2** [comment]
I'm not sure about this. The annotation is longer than the node above it, and with your method the annotation is written under the "def" but is cut off after three characters.

**Comment by snim2** [comment]
OK, the bug mentioned in the comment above is now fixed.

**Comment by snim2** [comment]
OK, I think this is ready for review...

File: lib/eco/treemanager.py [hunk 1]
```diff
@@ -526,6 +543,18 @@ def find_next(self):
         if self.last_search != "":
             self.find_text(self.last_search)
 
+    def find_text_no_cursor(self, text, parent_name='funcdef'):
```
Comment (ptersilie) [inline lib/eco/treemanager.py:546]: Is this still used?

Comment (snim2) [inline lib/eco/treemanager.py:546]: Good point. I'm not using it in this PR but it is likely to be used later on.

Comment (ptersilie) [inline lib/eco/treemanager.py:546]: Ok.

**Comment by ptersilie** [comment]
I think you forgot to add throbber.gif.

**Comment by snim2** [comment]
Added the throbber, when you have finished reviewing I'll squash that commit.

File: lib/eco/eco.py [hunk 2]
```diff
@@ -1042,5 +1064,49 @@ def run(self):
...
+        self.tab_bar = tab_bar
+        self.setAlignment(Qt.AlignCenter)
+        self._movie = QMovie("gui/throbber.gif")
+        self.setMovie(self._movie)
+
+    def hide(self):
+        """Hide throbber in left hand corner of tabbed notebook.
+        """
+        self._movie.stop()
+        self.tab_bar.setCornerWidget(None, corner=Qt.TopLeftCorner)
```
Comment (ptersilie) [inline lib/eco/eco.py]: It doesn't matter a great deal, but I think the waiting animation (which I quite like by the way) looks a bit better on the top right corner. What do you think?

**Comment by ptersilie** [comment]
Awesome, thanks. Rest looks good. If you squash I will merge. :)

**Comment by snim2** [comment]
Great, squashed!

**Pull request merged**